packages/firestore/index.rn.ts

```diff
@@ -0,0 +1,41 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import firebase from '@firebase/app';
+import { FirebaseNamespace } from '@firebase/app-types';
+
+import { Firestore } from './src/api/database';
+import { IndexedDbComponentProvider } from './src/core/component_provider';
+import { configureForFirebase } from './src/platform/config';
+
+import './register-module';
+import './src/platform_rn/rn_init';
+
+import { name, version } from './package.json';
+
+/**
+ * Registers the main Firestore ReactNative build with the components framework.
+ * Persistence can be enabled via `firebase.firestore().enablePersistence()`.
+ */
+export function registerFirestore(instance: FirebaseNamespace): void {
+  configureForFirebase(
+    instance,
+    (app, auth) => new Firestore(app, auth, new IndexedDbComponentProvider())
```

Comment:
RN doesn't support indexeddb, so we should use `MemoryComponentProvider` here. Probably just remove this file, and point `react-native` in the root package.json to the memory build.

Reply:
We have some users that are using our IndexedDb Shim support. I am not sure if we should break them in this PR. FWIW, our main Node build also ships with IndexedDb support.